src/plugins/visualizations/public/embeddable/visualize_embeddable.ts

```diff
@@ -465,4 +501,87 @@ export class VisualizeEmbeddable
       { showSaveModal: true, saveModalTitle }
     );
   };
+
+  /**
+   * Fetches any VisLayers, and filters out to only include ones in the list of
+   * input resource IDs, if specified. Assigns them to this.visLayers.
+   * Note this fn is public so we can fetch vislayers on demand when needed,
+   * e.g., generating other vis embeddables in the view events flyout.
+   */
+  public async populateVisLayers(): Promise<void> {
+    const visLayers = await this.fetchVisLayers();
+    this.visLayers =
+      this.visAugmenterConfig?.visLayerResourceIds === undefined
+        ? visLayers
+        : visLayers.filter((visLayer) =>
+            this.visAugmenterConfig?.visLayerResourceIds?.includes(visLayer.pluginResource.id)
```

Comment:
NIT: if line 514 is checking for vislayerresourceids and visaugmenterconfig to be undefined does it still need the optional operator on line 517?

Reply:
Good point, removed to look cleaner. I had added to appease the IDE warnings showing up but it looks like it's not understanding the check on line 514.